data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -48,23 +48,23 @@
 {% endif %}
 
 {% if has_start_date %}
-WHERE (COALESCE({{end_date}}, EXTRACT(DATE FROM {{end_datetime}}),
-    {{start_date}}, EXTRACT(DATE FROM {{start_datetime}})) >= d.deactivated_date
+WHERE (COALESCE(TIMESTAMP({{end_date}}), {{end_datetime}},
+    TIMESTAMP({{start_date}}), {{start_datetime}}) >= d.deactivated_datetime
 {% if table_ref.table_id == 'drug_exposure' %}
-OR verbatim_end_date >= d.deactivated_date)
+OR TIMESTAMP(verbatim_end_date) >= d.deactivated_datetime)
 {% else %} )
 {% endif %}
 {% elif table_ref.table_id == 'death' %}
-WHERE COALESCE(death_date, EXTRACT(DATE FROM death_datetime)) >= d.deactivated_date
+WHERE COALESCE(TIMESTAMP(death_date), death_datetime) >= d.deactivated_datetime
 {% elif table_ref.table_id in ['activity_summary', 'heart_rate_summary'] %}
-WHERE date >= d.deactivated_date
+WHERE TIMESTAMP(date) >= d.deactivated_datetime
 {% elif table_ref.table_id in ['heart_rate_minute_level', 'steps_intraday']  %}
-WHERE datetime >= PARSE_DATETIME('%F', CAST(d.deactivated_date as STRING))
+WHERE datetime >= PARSE_DATETIME('%F', CAST(d.deactivated_datetime as STRING))
 {% elif table_ref.table_id in ['drug_era', 'condition_era', 'dose_era', 'payer_plan_period', 'observation_period']  %}
-WHERE COALESCE(DATE({{table_ref.table_id + '_end_date'}}),
-DATE({{table_ref.table_id + '_start_date'}})) >= d.deactivated_date
+WHERE COALESCE(TIMESTAMP({{table_ref.table_id + '_end_date'}}),
+TIMESTAMP({{table_ref.table_id + '_start_date'}})) >= d.deactivated_datetime
 {% else %}
-WHERE COALESCE({{date}}, EXTRACT(DATE FROM {{datetime}})) >= d.deactivated_date
+WHERE COALESCE({{datetime}}, TIMESTAMP({{date}})) >= d.deactivated_datetime
```

Comment:
@hiro-mishima, @jp3477  - For this test to pass, I had to flip the order of the fields in the COALESCE on line 67.  Now that the comparison is based on a TIMESTAMP value, 3 test records should have been excluded as the deactivation time was 1:00:00 UTC.  When the created timestamp from date (TIMESTAMP({{date}})) is used as the primary field in the COALESCE, the comparison time for the records becomes 0:00:00 UTC, which is before the deactivation time and the records will not be excluded.  

It seems likely lines 51, 52, and 58 might need the same treatment, but I would welcome your input.

Reply:
Thanks for checking, @molly-giddings!

1. Casting Date type to DateTime type here may cause an incorrect condition check. I think this line should be updated to something like this:
`WHERE ({{datetime}} IS NOT NULL AND {{datetime}} >= d.deactivated_datetime)`
`OR ({{datetime}} IS NULL AND {{date}} >= DATE(d.deactivated_datetime))`

2. 51, 52, and 58 should be updated similarly.

3. For the same reason, 54, 60, and 64-65 should be Date vs. Date comparisons, not DateTime vs. DateTime.

4. Would it be possible to add more test data to `test_queries_to_retract_from_fake_dataset` to cover each of the patterns?

Reply:
@hiro-mishima , I have made changes for all of your suggestions.  I also had to break out the conditions starting on line 66 into two different groups as `drug_era`, `condition_era`, and `dose_era `now have start_date and end_date as TIMESTAMP rather than DATE.